data/txHandler.go

```diff
@@ -236,13 +254,24 @@ func (handler *TxHandler) backlogWorker() {
 			if !ok {
 				return
 			}
+			if wi.capguard != nil {
+				if err := wi.capguard.Release(); err != nil {
+					logging.Base().Warnf("Failed to release capacity to ElasticRateLimiter: %v", err)
+				}
+			}
 			if handler.checkAlreadyCommitted(wi) {
 				transactionMessagesAlreadyCommitted.Inc(nil)
+				if wi.capguard != nil {
+					wi.capguard.Served()
+				}
 				continue
 			}
 
 			// enqueue the task to the verification pool.
 			handler.txVerificationPool.EnqueueBacklog(handler.ctx, handler.asyncVerifySignature, wi, nil)
+			if wi.capguard != nil {
+				wi.capguard.Served()
+			}
```

Comment:
(1) I must be mistaken here. I don't know what I was thinking...
for (2 & 3), here is my suggestion. It will only add a single line for handling capguard.
If will clean up 30% of the code in this segment: 22 lines -> 14 lines 
Note: my suggestion has the changes to the file that came after this PR.

```
		case wi, ok := <-handler.backlogQueue:
			if !ok {
				// this is never happening since handler.backlogQueue is never closed
				return
			}
			ac := handler.checkAlreadyCommitted(wi)
			handleCapguard(wi, ac)
			if  ac {
				transactionMessagesAlreadyCommitted.Inc(nil)
				continue
			}
			// handler.streamVerifierChan does not receive if ctx is cancled
			select {
			case handler.streamVerifierChan <- &verify.UnverifiedElement{TxnGroup: wi.unverifiedTxGroup, BacklogMessage: wi}:
			case <-handler.ctx.Done():
				transactionMessagesDroppedFromBacklog.Inc(nil)
				return
			}
```

```
func handleCapguard(wi, alreadyCommitted) {
			if wi.capguard != nil {
				if err := wi.capguard.Release(); err != nil {
					logging.Base().Warnf("Failed to release capacity to ElasticRateLimiter: %v", err)
				}
			}
			if alreadyCommitted {
				if wi.capguard != nil {
					wi.capguard.Served()
				}
			}
}
```